File: user-list/db/userDAO.js
import pkg from "mongodb";
const { MongoClient } = pkg;

import Connection from "./connection.js";
import User from "../models/User.js";

/**
 * Definig user model
 */
export const userModel = {
    name: String,
    email: String,
    asignedTaskIds: Array
};


/**
 * Accepts an object of type User model and fetch matching data from DB
 * @param {User} userDocument 
 * @returns {boolean}
 */
export async function get(userDocument) {
    try {
        console.log(`Query data: ${JSON.stringify(userDocument)}`);
        let conn = new Connection();
        const connClient = await conn.getClient();

        let userCollection = connClient.client.db(conn.DB_NAME).collection("users");

        // userDocument is {} when all records to be fetched
        const result = userCollection.find(userDocument);

        if ((result.count()) === 0) {
            console.log("No documents found!");
        }
        let items = [];
        return result.forEach(item => items.push(item)).then(res => {
            conn.closeConnection();
            return {status: true, result: items};
        });
    } catch (error) {
        console.error("Error inserting user:", error);
        return false;
    }
};

File: user-delete/db/userDAO.py
from db.Connection import Connection

def delete_user(user_id: str) -> bool:
    """
    Accepts a user_id and deletes the user from the DB.
    :param user_id: str
    :return: bool
    """
    con = Connection()
    try:
        conn = con.get_connection()
        print("DEBUG")
        db = conn['client'][con.DB_NAME]
        user_collection = db["users"]
        query = {"_id": user_id}
        print(f"Query: {query}")
        result = user_collection.delete_one(query)
        con.close_connection()
        if result.deleted_count == 1:
            print("Successfully deleted one document.")
            return True
        else:
            print("No documents matched the query. Deleted 0 documents.")
            return False
    except Exception as e:
        print(e)
        # print(f"Error deleting user: {e}")
        con.close_connection()
        return False